sierra2mlir/src/compiler.rs

```diff
@@ -921,15 +899,15 @@ impl<'ctx> Compiler<'ctx> {
     pub fn op_llvm_gep<'a>(
         &self,
         block: &'a Block,
-        index: usize,
+        indexes: &[usize],
```

Comment:
Please correct me if I'm misunderstanding this, but aren't there specifically two indices for gep, one that indexes into the pointer as an array, and one that is to index into the element itself? If so could we have that as two named parameters to this function to avoid confusion?

Reply:
No, gep can have infinite (?) indexes and the dialect operation reflects that, by accepting many indices, it's just that when I initially added the gep function we only needed to index into structs, but now we need to index into arrays too, i think this signature is the most faithful to the operation.

https://llvm.org/docs/LangRef.html#getelementptr-instruction
https://mlir.llvm.org/docs/Dialects/LLVM/#llvmgetelementptr-mlirllvmgepop

Reply:
(the dynamic version of the gep which we have (accepts mlir Values) also has this signature)